frontend/src/features/dataset/DatasetPage.tsx

```diff
@@ -36,15 +36,35 @@ const DatasetPage: React.FC<{}> = ({
     dispatch(loadDataset(ref))
   }, [dispatch, qriRef.username, qriRef.name, qriRef.path, isNew])
 
+  const { pathname } = useLocation();
```

Comment:
any time we're considering calling `useLocation`, let's stop & think about routing.